Source/common/SNTMetricSetTest.m

```diff
@@ -0,0 +1,527 @@
+#import <XCTest/XCTest.h>
+
+#import "Source/common/SNTMetricSet.h"
+
+@interface SNTMetricCounterTest : XCTestCase
+@end
+
+@interface SNTMetricInt64GaugeTest : XCTestCase
+@end
+
+@interface SNTMetricDoubleGaugeTest : XCTestCase
+@end
+
+@interface SNTMetricBooleanGaugeTest : XCTestCase
+@end
+
+@interface SNTMetricStringGaugeTest : XCTestCase
+@end
+
+@interface SNTMetricSetTest : XCTestCase
+@end
+
+// Stub out NSDate's date method
```

Comment:
clever :)

You could also do with https://ocmock.org/reference/#mocking-class-methods if you needed finer control.